zhmcclient_mock/_urihandler.py

```diff
@@ -387,6 +396,23 @@ def check_set_noninput(method, uri, properties, prop_name, prop_value):
     properties[prop_name] = prop_value
 
 
+def check_invalid_query_parms(method, uri, query_parms, valid_query_parms):
+    """
+    Check that the query parameters are valid.
+
+    Raises:
+      BadRequestError with reason 1.
+    """
+    for qp in query_parms:
+        if qp not in valid_query_parms:
```

Comment:
Would it make sense to collect all invalid query params, and throw the exception after the loop, in case the set (string) with invalid query params isn't empty?
Why tell the user only about one problem at a time, when you could report all of them at once?

Reply:
DONE.